src/main/java/de/kekru/struktogrammeditor/control/Struktogramm.java

```diff
@@ -64,668 +66,805 @@
-			return -1;
+	// gibt die entsprechende Typnummer zu einen StruktogrammElement zurück
+	public static AnweisungsTyp strElementZuTyp(StruktogrammElement str) {
+
+		if (str instanceof Verzweigung) {// Wichtig: erst Verzweigung, dann Fallauswahl prüfen, weil Verzweigung von
+											// Fallauswahl erbt und somit eine Verzweigung bei (str instanceof
+											// Fallauswahl) true ergibt, selbiges auch mit Anweisung und
+											// Aussprung/Aufruf
+			return AnweisungsTyp.Verzweigung;
+		} else if (str instanceof Fallauswahl) {
+			return AnweisungsTyp.Fallauswahl;
+		} else if (str instanceof ForSchleife) {
+			return AnweisungsTyp.ForSchleife;
+		} else if (str instanceof WhileSchleife) {
+			return AnweisungsTyp.WhileSchleife;
+		} else if (str instanceof DoUntilSchleife) {
+			return AnweisungsTyp.DoUntilSchleife;
+		} else if (str instanceof Endlosschleife) {
+			return AnweisungsTyp.Endlosschleife;
+		} else if (str instanceof Aussprung) {
+			return AnweisungsTyp.Aussprung;
+		} else if (str instanceof Aufruf) {
+			return AnweisungsTyp.Aufruf;
+		} else if (str instanceof LeerElement) {
+			return AnweisungsTyp.LeerElement;
+		} else if (str instanceof Anweisung) {
+			return AnweisungsTyp.Anweisung;
+		} else {
+			return AnweisungsTyp.ERROR;
 		}
 	}
 
-
-
-
-	//erzeugt JScrollPane, legt dieses Struktogramm in das JScrollPane und gibt das JScrollPane zurück, damit es in das JTabbedPane gelegt werden kann (siehe StrTabbedPane.struktogrammHinzufuegen())
-	//	public JScrollPane gibScrollPaneFuerContainer(){
-	//		//		scrollpane = new JScrollPane(this);//http://www.dpunkt.de/java/Programmieren_mit_Java/Oberflaechenprogrammierung/14.html
-	//		//		scrollpane.setBounds(getBounds());
-	//		//
-	//		//		scrollpane.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS);
-	//		//		scrollpane.setHorizontalScrollBarPolicy(JScrollPane.HORIZONTAL_SCROLLBAR_ALWAYS);
-	//		//
-	//		//		setBounds(scrollpane.getBounds());
-	//		//
-	//		//		scrollpane.addComponentListener(new ComponentListener() { //Listener zum Mitverfolgen von Größenveränderungen durch den User
-	//		//			public void componentResized(ComponentEvent e){
-	//		//				if((g != null) && (ueberwacheResize)){//nur wenn die GUI ueberwacheResize true gesetzt hat, sonst wird es beim einfügen neuer Elemente ausgeführt, wenn der Anzeigebereich verlassen wird, und beim Scrollpane gibt es Probleme
-	//		//					zeichenbereichAktualisieren(); //wenn die Größe geändert wurde, muss der Zeichenbereich aktualisiert werden, sonst wird das Struktogramm gestreckt oder verkleinert dargestellt
-	//		//					zeichne();
-	//		//					ueberwacheResize = false;
-	//		//				}
-	//		//			}
-	//		//			public void componentMoved(ComponentEvent e){}
-	//		//			public void componentShown(ComponentEvent e){}
-	//		//			public void componentHidden(ComponentEvent e){}
-	//		//		});
-	//		//
-	//		//		return scrollpane;
+	// erzeugt JScrollPane, legt dieses Struktogramm in das JScrollPane und gibt das

```

Comment:
You can delete my old out commented code :)

Reply:
Alright